server/test/models/user.spec.js

```diff
@@ -0,0 +1,253 @@
+        ...userTest,
+        email: 'admin@gmail.com',
+        username: 'admin',
+        isAdmin: true
+      };
+      try {
+        newUser = await User.create(userTest1);
+      } catch (e) {
+        error = e;
+      } finally {
+        expect(error).to.equal(undefined);
+        expect(newUser.email).to.equal(userTest1.email);
+        expect(newUser.username).to.equal(userTest1.username);
+        expect(newUser.firstName).to.equal(userTest1.firstName);
+        expect(newUser.bio).to.equal(userTest1.bio);
+        expect(newUser.isAdmin).to.equal(true);
+      }
+    });
+
+    it('Should not allow duplicate emails', async () => {
+      let newUser;
+      let error;
+      const userTest1 = { ...userTest, username: 'lulu1' };
+      try {
+        newUser = await User.create(userTest1);
+      } catch (e) {
+        error = e;
+      } finally {
+        expect(newUser).to.equal(undefined);
+        expect(error).to.be.an.instanceof(Error);
+        expect(error.errors[0].message).to.equal('email must be unique');
+      }
+    });
+
+    it('Should not allow duplicate usernames', async () => {
+      let newUser;
+      let error;
+      const userTest1 = { ...userTest, email: 'lewislulu1@yahoo.com' };
+      try {
+        newUser = await User.create(userTest1);
+      } catch (e) {
+        error = e;
+      } finally {
+        expect(newUser).to.equal(undefined);
+        expect(error).to.be.an.instanceof(Error);
+        expect(error.errors[0].message).to.equal('username must be unique');
+      }
+    });
+
+    it('Should require email', async () => {
+      let newUser;
+      let error;
+      const userTest1 = { ...userTest, email: undefined };
+      try {
+        newUser = await User.create(userTest1);
+      } catch (e) {
+        error = e;
+      } finally {
+        expect(newUser).to.equal(undefined);
+        expect(error).to.be.an.instanceof(Error);
+        expect(error.errors[0].message).to.equal('Email is required.');
+      }
+    });
+
+    it('Should require username', async () => {
+      let newUser;
+      let error;
+      const userTest1 = { ...userTest, username: undefined };
+      try {
+        newUser = await User.create(userTest1);
+      } catch (e) {
+        error = e;
+      } finally {
+        expect(newUser).to.equal(undefined);
+        expect(error).to.be.an.instanceof(Error);
+        expect(error.errors[0].message).to.equal('Username is required.');
+      }
+    });
+
+    it('Should require last name', async () => {
+      let newUser;
+      let error;
+      const userTest1 = { ...userTest, lastName: undefined };
+      try {
+        newUser = await User.create(userTest1);
+      } catch (e) {
+        error = e;
+      } finally {
+        expect(newUser).to.equal(undefined);
+        expect(error).to.be.an.instanceof(Error);
+        expect(error.errors[0].message).to.equal('Last name is required.');
+      }
+    });
+
+    it('Should require last name', async () => {
```

Comment:
Well done @frankly034, I think this ```should require last name``` should be ```should require first name```.

Reply:
Ok. Thanks.